From 814a322a7dc443128fa79286c576c23798f5b1d3 Mon Sep 17 00:00:00 2001
From: Jeffy Chen <jeffy.chen@rock-chips.com>
Date: Mon, 9 Jul 2018 15:53:21 +0800
Subject: [PATCH] drm/rockchip: Sanity check fbdev_helper

The fbdev_helper would not be available when CONFIG_DRM_FBDEV_EMULATION
is disabled.

Change-Id: I9f79c170e862f24eb717c7940d8acaefd060740f
Signed-off-by: Jeffy Chen <jeffy.chen@rock-chips.com>
---
 drivers/gpu/drm/rockchip/rockchip_drm_drv.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/drivers/gpu/drm/rockchip/rockchip_drm_drv.c b/drivers/gpu/drm/rockchip/rockchip_drm_drv.c
index 4519d7250719..c24c6d907549 100644
--- a/drivers/gpu/drm/rockchip/rockchip_drm_drv.c
+++ b/drivers/gpu/drm/rockchip/rockchip_drm_drv.c
@@ -1485,6 +1485,9 @@ static int rockchip_drm_bind(struct device *dev)
 		struct drm_fb_helper *helper = private->fbdev_helper;
 		struct rockchip_crtc_state *s = NULL;
 
+		if (!helper)
+			break;
+
 		s = to_rockchip_crtc_state(crtc->state);
 		if (is_support_hotplug(s->output_type)) {
 			s->crtc_primary_fb = crtc->primary->fb;
-- 
2.35.3

